File: behaviorial/command/python/receiver.py
from abc import ABC, abstractmethod


class Device(ABC):
    @abstractmethod
    def on(self):
        pass

    @abstractmethod
    def off(self):
        pass

    @abstractmethod
    def is_on(self):
        pass


class Light(Device):
    def __init__(self):
        self._state = False

    def on(self):
        self._state = True
        print('The light is on')

    def off(self):
        self._state = False
        print('The light is off')

    def is_on(self):
        return self._state


class Fan(Device):
    def __init__(self):
        self._state = False

    def on(self):
        self._state = True
        print('The fan is on')

    def off(self):
        self._state = False
        print('The fan is off')

    def is_on(self):
        return self._state

File: behaviorial/command/typescript/receiver.ts
export interface Device {
	on(): void;
	off(): void;
	isOn(): boolean;
}

export class Light implements Device {
	private state: boolean = false;

	on(): void {
		this.state = true;
		console.log('The light is on');
	}

	off(): void {
		this.state = false;
		console.log('The light is off');
	}

	isOn(): boolean {
		return this.state;
	}
}

export class Fan implements Device {
	private state: boolean = false;

	on(): void {
		this.state = true;
		console.log('The fan is on');
	}

	off(): void {
		this.state = false;
		console.log('The fan is off');
	}

	isOn(): boolean {
		return this.state;
	}
}
